pkg/controller/podautoscaler/horizontal.go

```diff
@@ -172,8 +179,9 @@ func (a *HorizontalController) Run(ctx context.Context) {
 		return
 	}
 
-	// start a single worker (we may wish to start more in the future)
-	go wait.UntilWithContext(ctx, a.worker, time.Second)
+	for i := 0; i < workers; i++ {
+		go wait.UntilWithContext(ctx, a.worker, time.Second)
```

Comment:
I can see a potential problem with shuffled logs of different workers: log lines of different workers will be mixed up. At least we can pass a worker id and add for example `[Worker-%d] ...` (or `[W%d] ...` to save disk space) prefix for every klog call in `a.worker` to easily filter logs by worker.

Reply:
IMHO, I don't see this as a problem, namespace/name of the resource is appended to the log messages and the majority of k8s controllers is using the same pattern. To add some kind of index to the logs, I'd have to refactor the `wait.UntilWithContext()` and other funcitons, to add the index parameter there. Not sure if we should do this big change?

Reply:
Agree, each log line should rather contain namespace/name than worker thread id.